graylog2-server/src/main/java/org/graylog2/plugin/periodical/Periodical.java

```diff
@@ -39,12 +39,27 @@
     /**
      * Only start this thread on master nodes?
      *
-     * @return
+     * @deprecated Use {@link #leaderOnly()} instead.
+     */
+    @Deprecated
+    public boolean masterOnly() {
+        return false;
+    }
+
+    /**
+     * Determines if this periodical should run only on the leader node.
+     *
+     * @return {@code false} (default) if this periodical may run on every node. {@code true} if it may run only on the
+     * leader node
      */
-    public abstract boolean masterOnly();
+    public boolean leaderOnly() {
+        // Defaulting to the now deprecated original method to not break existing implementations.
+        return masterOnly();
```

Comment:
if this one returns false and `masterOnly()` calls `leaderOnly()` we have one /master/ less in our code ;)

Reply:
🤔  but all the existing periodicals out there might implement `masterOnly()` because they don't know anything about `leaderOnly()`. As we now have the `PeriodicalsService` evaluating `leaderOnly()` exclusively, we have to call the periodical's implementation of `masterOnly()` and not just assume `false`. Or am I making this too complicated?

Reply:
Sorry, I was distracted and should've read the comment. Makes sense to do it this way. Even though, I doubt that anyone implemented a master-only periodical.